custom_components/yahoofinance/sensor.py

```diff
@@ -179,7 +179,7 @@ def convert_timestamp_to_datetime(date_timestamp, return_format) -> str | None:
 
         date_timestamp = convert_to_float(date_timestamp)
         if date_timestamp is None or date_timestamp == 0:
-            return 0
+            return date_timestamp
```

Comment:
For most symbols, I am seeing `0` as the value for preMarketTime in data which appears as 0. 

![image](https://github.com/iprak/yahoofinance/assets/6459774/76c63ddb-affc-4fbc-9554-fd1d956b9311)

If we adjusted convert_timestamp_to_datetime to return None , then it would appear as Unknown as for 
dividendDate. Yahoo is not providing any data for dividendDate in this case.


Thoughts? If we changed convert_timestamp_to_datetime, then the unit tests would also need some adjustment.

Reply:
I agree that's why I changed the function to return the date_timestamp value. The float function returns either the numeric value which is either 0, epoch or None.if it is 0 or None then the if statement well return 0 or None. 

Shouldn't need any further changes.